### PR TITLE
[improve][txn][PIP-196]Segmented transaction buffer snapshot configuration

### PR DESCRIPTION
**Pull request by liangyepianzhou**
Master Issue: https://github.com/apache/pulsar/issues/16913
### Motivation
1. Add configurations.
2. Implement system topic client for snapshot topic and index topic.
3. Implement AbortedTxnProcessor.
4. Add transaction buffer snapshot metrics.
### Modification
Add configuration
### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by liangyepianzhou** [comment]
@codelipenghui @congbobo184 @gaoran10 Please help review this PR when you have time. THX.
@momo-jun Please help review this PR from a technical writer.

File: conf/broker.conf [hunk 1]
```diff
@@ -1497,12 +1497,22 @@ transactionCoordinatorEnabled=false
 transactionMetadataStoreProviderClassName=org.apache.pulsar.transaction.coordinator.impl.MLTransactionMetadataStoreProvider
 
 # Transaction buffer takes a snapshot after the number of transaction operations reaches this value.
+# If transaction buffer enables snapshot segment, transaction buffer updates snapshot metadata
+# after the number of transaction operations reaches this value.
 transactionBufferSnapshotMaxTransactionCount=1000
 
-# Transaction buffer take snapshot interval time
+# Transaction buffer take snapshot interval time.
```
Comment (momo-jun) [inline conf/broker.conf:1504]:
```suggestion
# The interval time for the transaction buffer to take snapshots.
```

File: conf/broker.conf [hunk 2]
```diff
@@ -1497,12 +1497,22 @@ transactionCoordinatorEnabled=false
...
+# If transaction buffer enables snapshot segment, this is transaction updating snapshot metadata internal time.
```
Comment (momo-jun) [inline conf/broker.conf:1505]:
```suggestion
# If the transaction buffer enables snapshot segment, it is the interval time for the transaction buffer to update snapshot metadata.
```

File: conf/broker.conf [hunk 3]
```diff
@@ -1497,12 +1497,22 @@ transactionCoordinatorEnabled=false
...
 # Unit : millisecond
 transactionBufferSnapshotMinTimeInMillis=5000
 
+# Whether to enable segmented transaction buffer snapshot which is able to handle a large number of aborted transactions.
```
Comment (momo-jun) [inline conf/broker.conf:1509]:
```suggestion
# Whether to enable segmented transaction buffer snapshot to handle a large number of aborted transactions.
```

File: conf/broker.conf [hunk 4]
```diff
@@ -1497,12 +1497,22 @@ transactionCoordinatorEnabled=false
...
+transactionBufferSegmentedSnapshotEnabled = false
+
+# Transaction buffer will stored the transaction ID of the aborted transaction, and take snapshot for it.
```
Comment (momo-jun) [inline conf/broker.conf:1512]:
```suggestion
# Transaction buffer stores the transaction ID of aborted transactions and takes snapshots.
```

File: conf/broker.conf [hunk 5]
```diff
@@ -1497,12 +1497,22 @@ transactionCoordinatorEnabled=false
...
+# This configuration is configured to determine the size of the snapshot segment. defualt:256kb.
```
Comment (momo-jun) [inline conf/broker.conf:1513]:
```suggestion
# This configuration determines the size of the snapshot segment. The default value is 256 KB (262144 bytes).
```

**Comment by congbobo184** [comment]
when https://lists.apache.org/thread/bqoy3oz8flvxy7xpmnw81cr4c9sz5vy0 +3 binding can merge

**Comment by liangyepianzhou** [comment]
> when https://lists.apache.org/thread/bqoy3oz8flvxy7xpmnw81cr4c9sz5vy0 +3 binding can merge

It already has +3 binding.

**Review by congbobo184 — APPROVED** [review]

File: conf/broker.conf [hunk 6]
```diff
@@ -1497,12 +1497,22 @@ transactionCoordinatorEnabled=false
...
 transactionBufferSnapshotMaxTransactionCount=1000
 
-# Transaction buffer take snapshot interval time
+# The interval time for transaction buffer to take snapshots.
+# If transaction buffer enables snapshot segment, it is the interval time for transaction buffer to update snapshot metadata.
 # Unit : millisecond
 transactionBufferSnapshotMinTimeInMillis=5000
 
+# Whether to enable segmented transaction buffer snapshot to handle a large number of aborted transactions.
+transactionBufferSegmentedSnapshotEnabled = false
```
Comment (codelipenghui) [inline conf/broker.conf:1510]:
```suggestion
transactionBufferSegmentedSnapshotEnabled=false
```

File: conf/standalone.conf [hunk 7]
```diff
@@ -1108,12 +1108,22 @@ transactionCoordinatorEnabled=false
...
 transactionBufferSnapshotMaxTransactionCount=1000
 
-# Transaction buffer take snapshot interval time
+# The interval time for transaction buffer to take snapshots.
+# If transaction buffer enables snapshot segment, it is the interval time for transaction buffer to update snapshot metadata.
 # Unit : millisecond
 transactionBufferSnapshotMinTimeInMillis=5000
 
+# Whether to enable segmented transaction buffer snapshot to handle a large number of aborted transactions.
+transactionBufferSegmentedSnapshotEnabled = false
```
Comment (codelipenghui) [inline conf/standalone.conf:1121]:
```suggestion
transactionBufferSegmentedSnapshotEnabled=false
```

**Review by codelipenghui — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**